src/services/mongoose/wallets.ts

```diff
@@ -12,27 +13,57 @@ export const WalletsRepository = (): IWalletsRepository => {
       if (!result) {
         return new CouldNotFindError()
       }
-      const onChainAddressIdentifiers = result.onchain.map(({ pubkey, address }) => {
-        return {
-          pubkey: pubkey as Pubkey,
-          address: address as OnChainAddress,
-        }
-      })
-      const onChainAddresses = () =>
-        onChainAddressIdentifiers.map(({ address }) => address)
-
-      return {
-        id: walletId,
-        depositFeeRatio: result.depositFeeRatio,
-        onChainAddresses,
-        onChainAddressIdentifiers,
+      return resultToWallet(result)
+    } catch (err) {
+      return new UnknownRepositoryError(err)
+    }
+  }
+
+  const findByWalletname = async (
+    username: Walletname,
+  ): Promise<Wallet | RepositoryError> => {
+    try {
+      const result = await User.findOne({ username: caseInsensitiveRegex(username) })
+      if (!result) {
+        return new CouldNotFindError()
       }
+
+      return resultToWallet(result)
     } catch (err) {
       return new UnknownRepositoryError(err)
     }
   }
 
   return {
     findById,
+    findByWalletname,
+  }
+}
+
+const resultToWallet = (result: UserType): Wallet => {
+  const walletId = result.id as WalletId
+
+  const walletname: Walletname = result.username
+    ? (result.username as Walletname)
+    : ("" as Walletname)
```

Comment:
from talking with @samerbuna about this a while ago I think using `null` for absence of a string is probably better.

Reply:
Done here: https://github.com/GaloyMoney/galoy/pull/475/commits/beec2a611a6434636fe0ae7d9ba5f59214b9bec7